src/app/services/transaction/transaction-base/transaction-base.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component } from "@angular/core";
+import { IgxTransactionService, State, Transaction, TransactionType } from "igniteui-angular";
+import { IItem, WISHLIST } from "../data";
+
+@Component({
+    selector: "transaction-base",
+    styleUrls: ["./transaction-base.component.scss"],
+    templateUrl: "transaction-base.component.html"
+})
+export class TransactionBaseComponent {
+    public wishlist: IItem[];
+    public name: string;
+    public price: string;
+    public addDisabled: boolean;
+    public deleteDisabled: boolean;
+    public editDisabled: boolean;
+
+    constructor(public transactions: IgxTransactionService<Transaction, State>) {
+        this.wishlist = WISHLIST;
+    }
+
+    public onAdd(event): void {
+        const item: IItem = { id: 4, name: "Yacht", price: "A lot!" };
+        this.transactions.add({ id: 4, type: TransactionType.ADD, newValue: item });
+
+        /** visualization */
+        this.addDisabled = true;
+        this.name = item.name;
+        this.price = item.price;
+    }
+
+    public onEdit(event): void {
+        const newPrice = "54$";
+        this.transactions.add({
+            id: 2, type: TransactionType.UPDATE, newValue: { name: "Apple", price: newPrice }
+        }, this.wishlist[1]);
+
+        /** visualization */
+        this.editDisabled = true;
+        this.name = this.wishlist[1].name;
+        this.price = newPrice;
+    }
+
+    public onDelete(event): void {
+        /** visualization */
```

Comment:
The /** visualization */ may be below the transaction itself like in the rest of the methods.